OpenRA.Mods.Cnc/Traits/World/TSEditorResourceLayer.cs

```diff
@@ -50,30 +52,30 @@ bool IsValidVeinNeighbour(CPos cell, CPos neighbour)
 				return true;
 
 			// Neighbour must be flat or a cardinal slope, unless the resource cell itself is a slope
-			if (Map.Ramp[cell] == 0 && Map.Ramp[neighbour] > 4)
+			if (mapRamp[cell] == 0 && mapRamp[neighbour] > 4)
 				return false;
 
 			var terrainInfo = Map.Rules.TerrainInfo;
-			var terrainType = terrainInfo.TerrainTypes[terrainInfo.GetTerrainInfo(Map.Tiles[neighbour]).TerrainType].Type;
+			var terrainType = terrainInfo.TerrainTypes[terrainInfo.GetTerrainInfo(((IMapTiles)Map).Tiles[neighbour]).TerrainType].Type;
 			return info.ResourceTypes[info.VeinType].AllowedTerrainTypes.Contains(terrainType);
 		}
 
 		protected override bool AllowResourceAt(string resourceType, CPos cell)
 		{
-			var mapResources = Map.Resources;
+			var mapResources = ((IMapResource)Map).Resources;
```

Comment:
`map.GetResourceManager()?.Resources`

Reply:
See my comment above about `IMapTiles`, this functionality is also no longer in `Map`, and we want this functionality to be behind the interface `IMapResource` so that map formats have the choice on whether to implement it or not.